packages/nimble-components/src/tree-item/template.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    children,
+    elements,
+    html,
+    ref,
+    slotted,
+    ViewTemplate,
+    when
+} from '@microsoft/fast-element';
+import {
+    TreeItemOptions,
+    FoundationElementTemplate,
+    startSlotTemplate,
+    endSlotTemplate
+} from '@microsoft/fast-foundation';
+import type { TreeItem } from '.';
+
+// prettier-ignore
+export const template: FoundationElementTemplate<
+ViewTemplate<TreeItem>,
+TreeItemOptions
+> = (context, definition) => html<TreeItem>`
+    <template
+        role="treeitem"
+        slot="${x => (x.isNestedItem() ? 'item' : null)}"
+        tabindex="-1"
+        ?group-selected="${x => x.isGroupSelected()}"
+        class="${x => (x.expanded ? 'expanded' : '')} ${x => (x.selected ? 'selected' : '')} ${x => (x.nested ? 'nested' : '')}
+            ${x => (x.disabled ? 'disabled' : '')}"
+        aria-expanded="${x => (x.childItems && x.childItemLength() > 0 ? x.expanded : null)}"
+        aria-selected="${x => x.selected}"
+        aria-disabled="${x => x.disabled}"
+        @focusin="${(x, c) => x.handleFocus(c.event as FocusEvent)}"
+        @focusout="${(x, c) => x.handleBlur(c.event as FocusEvent)}"
+        ${children({ property: 'childItems', filter: elements() })}
```

Comment:
My thought from our earlier discussions was  that we can have a single `@children()` on the tree view and [leverage the subtree option](https://www.fast.design/docs/fast-element/using-directives#the-children-directive): 
![image](https://user-images.githubusercontent.com/1588923/232160546-f16f2688-e94e-4ecc-9797-df817b67f49d.png)

to find the possible child tree-items, and subscribe to them for selection changes. While the approach done here is viable, it's doing a lot more observation and subscription management through the layers so curious if that is needed.

Reply:
Sorry, I probably should have specified that I tried to add `${children({ property: 'selectedDescendants', subtree: true, selector: [selected] })}`, but apparently you can't put two `children` directives on the same element. The second one just gets ignored.

If you're suggesting that I modify the existing directive to get all descendants instead of just direct children, that would break a lot of existing logic that depends on `childItems` having just the direct children. I guess I might be able to maintain a separate filtered array, though... Is that what you had in mind?

Reply:
We're now using a `children` directive on the tree view that gets all the selected descendants. Whenver this changes, we clear any existing `group-selected` attributes and recalculate.